src/verify.c

```diff
@@ -959,28 +1004,79 @@ enum swupd_code verify_main(void)
 		}
 	}
 
-	ret = add_included_manifests(official_manifest, &subs);
+	/* get the list of subscribed (installed) bundles */
+	read_subscriptions(&all_subs);
+	selected_subs = all_subs;
+	if (cmdline_option_bundles) {
```

Comment:
you can group this block in a function to validate the bundle names and print all needed messages returning just a boolean to say if we should continue or abort.
we can free any resource that we don't need anymore and keep all tmp variables inside the function

Reply:
it is not that easy in this case, the loop not only validate the names of the bundles provided but it also creates the subscription in the appropriate list, so I could extract this block t a function but the function would be like:
`validate_provided_bundles(cmdline_option_bundles, official_manifest, all_subs, bundles_subs);`
I don't see much benefit out of it, we would still need to keep all variables in the verify_main function and we won't free any resource at this point anyway.
Ohh and the function could not return a bool, since it'd need to return 2 values:
- the boolean go/no go result
- a boolean value to indicate there was at least one invalid bundle specified by the user, we need that value later in the code

Bottom line I don't see this saving any resources, or even making the code more readable, IMO it just makes it more complex